ckan/cli/file.py

```diff
@@ -0,0 +1,567 @@
+        click.secho(f"\t{content_type}: {click.style(count, bold=True)}")
+
+
+@stats.command("owner")
+@storage_option
+@click.option(
+    "-v",
+    "--verbose",
+    is_flag=True,
+    help="Show distribution for every owner ID",
+)
+def stats_owner(storage_name: str | None, verbose: bool):
+    """Files distribution by owner."""
+    storage_name = storage_name or config["ckan.files.default_storages.default"]
+    owner_col = (
+        sa.func.concat(model.Owner.owner_type, " ", model.Owner.owner_id)
+        if verbose
+        else sa.func.concat(model.Owner.owner_type, "")
+    )
+
+    stmt = (
+        sa.select(
+            owner_col.label("owner"),
+            sa.func.count(model.File.id),
+        )
+        .where(model.File.storage == storage_name)
+        .outerjoin(
+            model.Owner,
+            sa.and_(
+                model.Owner.item_id == model.File.id,
+                model.Owner.item_type == "file",
+            ),
+        )
+        .group_by(owner_col)
+    ).order_by(owner_col)
+
+    total = model.Session.scalar(sa.select(sa.func.sum(stmt.c.count)))
+    click.secho(
+        f"Storage {click.style(storage_name, bold=True)} contains "
+        + f"{click.style(total, bold=True)} files",
+    )
+    for owner, count in model.Session.execute(stmt):
+        clean_owner = owner.strip() or click.style(
+            "has no owner",
+            underline=True,
+            bold=True,
+        )
+        click.secho(
+            f"\t{clean_owner}: {click.style(count, bold=True)}",
+        )
+
+
+@file.group()
+def maintain():
+    """Storage maintenance."""
+
+
+@maintain.command()
+@storage_option
+@click.option("--remove", is_flag=True, help="Remove files")
+def empty_owner(storage_name: str | None, remove: bool):
+    """Manage files that have no owner."""
+    storage_name = storage_name or config["ckan.files.default_storages.default"]
+    try:
+        storage = files.get_storage(storage_name)
+    except files.exc.UnknownStorageError as err:
+        error_shout(err)
+        raise click.Abort from err
+
+    if remove and not storage.supports(files.Capability.REMOVE):
+        error_shout(f"Storage {storage_name} does not support file removal")
+        raise click.Abort
+
+    stmt = (
+        sa.select(model.File)
+        .outerjoin(model.File.owner)
+        .where(model.File.storage == storage_name, model.Owner.owner_id.is_(None))
+    )
+
+    total = model.Session.scalar(stmt.with_only_columns(sa.func.count()))
+    if not total:
+        click.echo(f"Every file in storage {storage_name} has owner reference")
+        return
+
+    click.echo("Following files do not have owner reference:")
+
+    for file in model.Session.scalars(stmt):
+        size = fk.humanize_filesize(file.size)
+        click.echo(f"\t{file.id}: {file.name} [{file.content_type}, {size}]")
+
+    if remove and click.confirm("Do you want to delete these files?"):
```

Comment:
good to have a `yes` option here too for consistency

Reply:
I followed your advice above and replaced all confirmations with command flags. So you either specify the flag like `--remove` and files will be removed, or run command without this flag and see only statistics(with a message that explains, how they can be removed)